# marauders: name=coalesce_contiguous_d1999f4
diff --git a/src/map.rs b/src/map.rs
--- a/src/map.rs
+++ b/src/map.rs
@@ -276,14 +276,14 @@
         // whose start is less than or equal to the start of the range to insert,
         // or the one before that if both of the above cases exist.
         let mut candidates =         /* marauders:variation=coalesce_contiguous;tags= */
-        self
-                    .btm
-                    .range::<RangeStartWrapper<K>, (Bound<&RangeStartWrapper<K>>, Bound<&RangeStartWrapper<K>>)>((
-                        Bound::Unbounded,
-                        Bound::Included(&new_start_wrapper),
-                    ))
-                    .rev()
-                    .take(2)
+        self
+                    .btm
+                    .range::<RangeStartWrapper<K>, (Bound<&RangeStartWrapper<K>>, Bound<&RangeStartWrapper<K>>)>((
+                        Bound::Unbounded,
+                        Bound::Included(&new_start_wrapper),
+                    ))
+                    .rev()
+                    .take(1)
             .filter(|(stored_start_wrapper, _stored_value)| {
                 // Does the candidate range either overlap
                 // or immediately precede the range to insert?
